src/parsing.jl

```diff
@@ -244,6 +244,37 @@ end
     return block
 end
 
+###
+### Buses
+###
+
+@generated function parse_row!(rec::R, bytes, pos, len, options) where {R <: Buses30}
+    block = Expr(:block)
+    n = fieldcount(R)
+    for col in 1:(n - 2)
```

Comment:
we need to handle the last _two_ entires differently, becase the second-last entry (the last "real" column) we need to be sure to stop parsing _before_ the comment

Reply:
Sorry, I'm unfamiliar with the package, so I have some potentially basic questions.

I see that this function dispatches over `Buses30`. But it's not always the case that a PSSE v30 will have the comment. How do we deal with both situations? Also, it's not just the buses that have augmented names, other components have it too; are they being accounted for?

Reply:
> it's not always the case that a PSSE v30 will have the comment. How do we deal with both situations?

we define `Buses30` as having 12 columns, with the 12th being the comments. In this function we parse the first (12 - 2) = 10 columns as usual. We then parse the 11th (and potentially last column present, if no comments) with whitespace as the delimiter (i.e. stop parsing when we hit a space or newline). If when parsing the 11th column we hit the newline character then there are no comments and we set the 12th column as `missing`, else if we hit whitespace we then parse the rest of the line as the 12th and final column. See the `if newline` in the code.

> Also, it's not just the buses that have augmented names, other components have it too; are they being accounted for?

As far as i understood only the Buses comments ("augmented names") were needed, so this PR only parses comments from the Buses section (and only for v30 files). I doesn't extract comments present in any other section. Do we need to?

These are good question and i'm very happy to answer as many questions as you have!

Reply:
oh, wait, for other components, these comments can be reconstructed from columns already in the data, no?

Reply:
transformers are the biggest pain 😂